fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1351,20 +1351,29 @@ public boolean shouldAllowTypeConversionContinue(IndexBuildProto.IndexBuildIndex
                 }
             }
             if (newMethod == IndexBuildProto.IndexBuildIndexingStamp.Method.MUTUAL_BY_RECORDS) {
-                if (oldMethod == IndexBuildProto.IndexBuildIndexingStamp.Method.MULTI_TARGET_BY_RECORDS ||
-                        oldMethod == IndexBuildProto.IndexBuildIndexingStamp.Method.BY_RECORDS) {
+                if (!isTypeConversionAllowed(TakeoverTypes.BY_RECORDS_TO_MUTUAL)) {
+                    // This conversion is not allowed, check no further
+                    return false;
+                }
+                if (oldMethod == IndexBuildProto.IndexBuildIndexingStamp.Method.MULTI_TARGET_BY_RECORDS) {
+                    // This conversion should be allowed if:
+                    // a) The new mutual indexing set is identical to the old one
+                    // b) The mutual indexing continues only one target index (splitting it to one at a tiem)
+                    return newStamp.getTargetIndexCount() == 1 ||
+                            (newStamp.getTargetIndexCount() == oldStamp.getTargetIndexCount() &&
+                                    new HashSet<>(newStamp.getTargetIndexList()).containsAll(oldStamp.getTargetIndexList()));
+                }
+
+                if (oldMethod == IndexBuildProto.IndexBuildIndexingStamp.Method.BY_RECORDS) {
                     // Note that single-target-mutual is a private case of multi-target-mutual
-                    return isTypeConversionAllowed(TakeoverTypes.BY_RECORDS_TO_MUTUAL);
+                    return newStamp.getTargetIndexCount() == 1; // No real need to check the stamp's index name
```

Comment:
What happens if the indexes are different?

Reply:
The old stamp does not have a target index (sort of backward compatibility, maybe it should be changed in the future). 
The new stamp is generated by the mutual indexer. If we check the stamp's validity here, we should check it everywhere.